src/execution/physical_plan/plan_comparison_join.cpp

```diff
@@ -200,6 +202,82 @@ unique_ptr<PhysicalOperator> PhysicalPlanGenerator::CreatePlan(LogicalComparison
 		return make_unique<PhysicalCrossProduct>(op.types, std::move(left), std::move(right), op.estimated_cardinality);
 	}
 
+	if (op.join_reftype == JoinRefType::ASOF) {
+		//	Validate
+		vector<idx_t> equi_indexes;
```

Comment:
I wonder if it makes sense for the ASOF join to be a separate `LogicalOperator` similar to the `LogicalPositionalJoin` - e.g. a `LogicalASOFJoin`. It seems like the ASOF join code is mostly disjoint from the rest of the join code, and merging it into the `LogicalComparisonJoin` requires taking extra care during optimizations and such.

Reply:
Yeah I thought about that and decided not to for some reason. But it was earlier in the process so I should probably revisit it.

Reply:
It touched a lot of code, but it's probably the Right Thing To Do.